File: springboot-service/src/main/java/com/service/app/dto/MessageDto.java
package com.service.app.dto;

import com.service.app.entity.Message;
import com.service.app.enums.SourceType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;

@Getter
@Setter
@NoArgsConstructor
public class MessageDto {
    private String conversation;
    private String sourceType;
    private long messageNumber;
    private OffsetDateTime timestamp;

    public MessageDto(Message message){
        this.conversation = message.getConversation();
        this.sourceType = message.getSourceType().getName();
        this.messageNumber = message.getMessageNumber();
        this.timestamp = message.getTimestamp().atOffset(ZoneOffset.UTC);
    }

    public Message toMessage(){
        return new Message(conversation, SourceType.fromName(sourceType), messageNumber, timestamp.toInstant());
    }
}

File: dotnet-service/src/DTOs/MessageDTO.cs
using dotnet_service.Entities;
using dotnet_service.Enums;

namespace dotnet_service.DTOs;

public class MessageDTO
{
    public string Source { get; set; }
    public string Conversation { get; set; }
    public long MessageNumber { get; set; }
    public DateTime Timestamp { get; set; }
    
    public MessageDTO(Message message)
    {
        Source = message.Source.ToName();
        Conversation = message.Conversation;
        MessageNumber = message.MessageNumber;
        Timestamp = message.Timestamp;
    }

    public Message ToMessage()
    {
        return new Message
        {
            MessageNumber = MessageNumber,
            Conversation = Conversation,
            Source = SourceTypeUtils.FromName(Source),
            Timestamp = Timestamp
        };
    }
}